amd/src/connectivity_test.js

```diff
@@ -0,0 +1,32 @@
+
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Module to check connectivity to IdP endpoint from client.
+ *
+ * @module     auth_saml2/connectivity_test
+ * @copyright  2023 Peter Burnett <peterburnett@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+export const init = (checkTarget, redirURL) => {
+    var http = new XMLHttpRequest();
+    http.open('HEAD', checkTarget);
+    http.addEventListener('load', () => {
```

Comment:
If the check fails can we properly handle this? Is it worth setting something in the session so we don't repeat it or is this not a big deal in practice?

Reply:
Im not too worried about this, because if it fails for something transient, we would want it to rerun, and once a user is authed, its moot anyway.